src/translatador.cpp

```diff
@@ -0,0 +1,385 @@
+﻿#include <translatador.h>
+
+#include <marian.h>
+#include <common/options.h>
+#include <data/types.h>
+#include <translator/annotation.h>
+#include <translator/beam_search.h>
+#include <translator/definitions.h>
+#include <translator/logging.h>
+#include <translator/text_processor.h>
+#include <variant>
+
+namespace bergamot = marian::bergamot;
+
+static std::mutex init_mutex;
+static bool initialized;
+
+static thread_local std::string last_error;
+
+void initialize() {
+    std::lock_guard guard(init_mutex);
+    if (!initialized) {
+        marian::setThrowExceptionOnAbort(true);
+        for (const Logger& logger : createLoggers()) {
+            logger->set_level(spdlog::level::level_enum::off);
+        }
+        initialized = true;
+    }
+}
+
+struct Buffer {
```

Comment:
Does this own `data`? if it does, this should definitely have a move constructor.

Reply:
It does not own `data` 🙂